graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -983,6 +1128,34 @@ int64_t Pick(
   }
 }
 
+template <typename PickedType>
+int64_t TemporalPick(
+    torch::Tensor seed_timestamp, torch::Tensor csc_indices,
+    int64_t seed_offset, int64_t offset, int64_t num_neighbors, int64_t fanout,
+    bool replace, const torch::TensorOptions& options,
+    const torch::optional<torch::Tensor>& probs_or_mask,
+    const torch::optional<torch::Tensor>& node_timestamp,
+    const torch::optional<torch::Tensor>& edge_timestamp,
+    PickedType* picked_data_ptr) {
+  auto mask = TemporalMask(
+      seed_timestamp[seed_offset].item<int64_t>(), csc_indices, probs_or_mask,
+      node_timestamp, edge_timestamp, {offset, offset + num_neighbors});
+  torch::Tensor masked_prob;
+  if (probs_or_mask.has_value()) {
+    masked_prob =
+        probs_or_mask.value().slice(0, offset, offset + num_neighbors) * mask;
+  } else {
+    masked_prob = mask.to(torch::kFloat32);
+  }
+  auto picked_indices = NonUniformPickOp(masked_prob, fanout, replace);
+  auto picked_indices_ptr = picked_indices.data_ptr<int64_t>();
+  for (int i = 0; i < picked_indices.numel(); ++i) {
+    picked_data_ptr[i] =
```

Comment:
Why not use tensor op here to leverage the underlying vectolization optimization?

Reply:
Do you mean we can use `picked_data_ptr` as a tensor? I just follow other sampling interface here and not sure why we need to pass value to return by a raw pointer.